internal/codespaces/connection/port_forwarder.go

```diff
@@ -0,0 +1,243 @@
+package connection
```

Comment:
Just a thought, but should this live in its own package?

Reply:
It definitely can! I went back and forth on it since it's pretty tightly coupled with the connection stack, but I think it is probably better as its own for readability.